[CI] DataLifecycleServiceIT testAutomaticForceMerge failing

It failed on my PR. But I see it has failed on main as well. It does not reproduce. But since it is a new test, maybe there is something that experts can take a closer look.

**Build scan:**
https://gradle-enterprise.elastic.co/s/hwqc6m4l3z5s2/tests/:modules:dlm:internalClusterTest/org.elasticsearch.dlm.DataLifecycleServiceIT/testAutomaticForceMerge

**Reproduction line:**
```
./gradlew ':modules:dlm:internalClusterTest' --tests "org.elasticsearch.dlm.DataLifecycleServiceIT.testAutomaticForceMerge" -Dtests.seed=73A7958A5BFF3FAF -Dtests.locale=es -Dtests.timezone=Africa/Bissau -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.dlm.DataLifecycleServiceIT&tests.test=testAutomaticForceMerge

**Failure excerpt:**
```
java.lang.AssertionError: The segments for .ds-metrics-foo-2023.05.15-000001 were not merged
Expected: a value less than <1>
     but: <1> was equal to <1>

  at __randomizedtesting.SeedInfo.seed([73A7958A5BFF3FAF:87D280C491CE4F45]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.elasticsearch.dlm.DataLifecycleServiceIT.lambda$testAutomaticForceMerge$6(DataLifecycleServiceIT.java:343)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1158)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1131)
  at org.elasticsearch.dlm.DataLifecycleServiceIT.testAutomaticForceMerge(DataLifecycleServiceIT.java:325)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

Also at https://gradle-enterprise.elastic.co/s/kfx7tvtpvewqk/tests/:modules:dlm:internalClusterTest/org.elasticsearch.dlm.DataLifecycleServiceIT/testAutomaticForceMerge?top-execution=1 

```
java.lang.AssertionError: The segments for .ds-metrics-foo-2023.05.15-000001 were not merged |  
-- | --
  | Expected: a value less than <4> |  
  | but: <4> was equal to <4> |  

...

REPRODUCE WITH: ./gradlew ':modules:dlm:internalClusterTest' --tests "org.elasticsearch.dlm.DataLifecycleServiceIT.testAutomaticForceMerge" -Dtests.seed=9E737181F8C2F7E1 -Dtests.locale=pl -Dtests.timezone=Africa/Bamako -Druntime.java=17
```

Will mute.

--- a/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
+++ b/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java
@@ -260,7 +260,6 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
         });
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/96084")
     public void testAutomaticForceMerge() throws Exception {
         /*
          * This test makes sure that (1) DLM does _not_ call forcemerge on an index in the same DLM pass when it rolls over the index and
@@ -295,6 +294,20 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
                 FlushResponse flushResponse = client().admin().indices().flush(new FlushRequest(toBeRolledOverIndex)).actionGet();
                 assertThat(flushResponse.getStatus(), equalTo(RestStatus.OK));
             }
+
+            final String toBeForceMergedIndex;
+            final int preDlmSegmentsForceMergedIndex;
+
+            if (currentGeneration == 1) {
+                toBeForceMergedIndex = null; // Not going to be used
+                preDlmSegmentsForceMergedIndex = -1; // Not going to be used
+            } else {
+                toBeForceMergedIndex = DataStream.getDefaultBackingIndexName(dataStreamName, currentGeneration - 1);
+                preDlmSegmentsForceMergedIndex = getSegmentCount(toBeForceMergedIndex);
+                logger.info("preDlmSegmentsForceMergedIndex: {}", preDlmSegmentsForceMergedIndex);
+            }
+            final int preDlmSegmentsAboutToBeRolledOverIndex = getSegmentCount(toBeRolledOverIndex);
+            logger.info("preDlmSegmentsAboutToBeRolledOverIndex: {}", preDlmSegmentsAboutToBeRolledOverIndex);
             /*
              * Without the following, calls to forcemerge are essentially a no-op since it has already done automatic merging. Setting
              * merge_factor on its own does not do anything, but it results in calls to forcemerge making observable changes to the
@@ -304,17 +317,6 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
                 Settings.builder().put(MergePolicyConfig.INDEX_MERGE_POLICY_MERGE_FACTOR_SETTING.getKey(), 5),
                 toBeRolledOverIndex
             );
-
-            final String toBeForceMergedIndex;
-            final int preDlmSegmentsForceMergedIndex;
-            if (currentGeneration == 1) {
-                toBeForceMergedIndex = null; // Not going to be used
-                preDlmSegmentsForceMergedIndex = -1; // Not going to be used
-            } else {
-                toBeForceMergedIndex = DataStream.getDefaultBackingIndexName(dataStreamName, currentGeneration - 1);
-                preDlmSegmentsForceMergedIndex = getSegmentCount(toBeForceMergedIndex);
-            }
-            final int preDlmSegmentsAboutToBeRolledOverIndex = getSegmentCount(toBeRolledOverIndex);
             int currentBackingIndexCount = currentGeneration;
             DataLifecycleService dataLifecycleService = internalCluster().getInstance(
                 DataLifecycleService.class,
@@ -346,6 +348,7 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
                         postDlmSegmentsForceMergedIndex,
                         lessThan(preDlmSegmentsForceMergedIndex)
                     );
+                    logger.info("postDlmSegmentsForceMergedIndex: {}", postDlmSegmentsForceMergedIndex);
                 }
                 // We want to assert that when DLM rolls over the write index it, it doesn't forcemerge it on that iteration:
                 assertThat(
@@ -355,10 +358,6 @@ public class DataLifecycleServiceIT extends ESIntegTestCase {
                 );
             });
         }
-    }
-
-    private static void enableDLM(TimeValue pollInterval) {
-        updateClusterSettings(Settings.builder().put(DataLifecycleService.DLM_POLL_INTERVAL, pollInterval));
     }
 
     private static void disableDLM() {

